engines/instance_verification/lib/instance_verification/engine.rb

```diff
@@ -15,16 +15,23 @@ def self.update_cache(cache_entry, mode, registry: false)
     )
   end
 
-  def self.build_cache_entry(remote_ip, system_login, encoded_reg_code, mode, product)
+  def self.build_cache_entry(remote_ip, system_login, params, mode, product)
     if mode == 'payg'
       [remote_ip, system_login, product.id].join('-')
     elsif mode == 'registry'
       [remote_ip, system_login].join('-')
     else
       # for byos or hybrid cache
+      instance_data = params.fetch(:instance_data, '')
+      iid = if instance_data.present?
+              InstanceVerification.provider.new(nil, nil, nil, instance_data).instance_identifier
+            else
+              ''
+            end
```

Comment:
Optional: Since you're returning `''` you can use the `Object#presence` method, so you don't need to use a if-else statement.

```suggestion
      iid =  InstanceVerification.provider.new(nil, nil, nil, instance_data).instance_identifier.presence || ''
```

Reply:
the issue/if condition is if `instance_data` is `nil` or present, not `instance_identifier`

Reply:
You're right. 
I get it wrong